Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Sharing/MarkerDetectorSpatialLocalizer.cs

```diff
@@ -30,47 +37,106 @@ private void OnValidate()
 
         private void Awake()
         {
-            DebugLog("Awake", Guid.Empty);
+            DebugLog("Awake");
             markerDetector = MarkerDetector as IMarkerDetector;
             if (markerDetector == null)
             {
                 Debug.LogWarning("Marker detector not appropriately set for MarkerDetectorSpatialLocalizer");
             }
+        }
+
+        public override bool TryDeserializeSettings(BinaryReader reader, out MarkerDetectorLocalizationSettings settings)
+        {
+            return MarkerDetectorLocalizationSettings.TryDeserialize(reader, out settings);
+        }
 
-            spatialCoordinateService = new MarkerDetectorCoordinateService(markerDetector, debugLogging);
+        public override ISpatialLocalizationSession CreateLocalizationSession(MarkerDetectorLocalizationSettings settings)
+        {
+            return new LocalizationSession(this, settings);
         }
```

Comment:
I think we need to check whether a session already exists/is localizing compared to blindly creating them. With how this exists today, you may end up having multiple sessions using the same marker detector, which is going to fail to find all markers with how marker detectors are started/stopped currently.

Reply:
That can't happen today as long as localization is controlled through the SpatialCoordinateSystemManager, which allows only one session to happen at a time

Reply:
And I think some central service/authority has to be the gatekeeper for how sessions are managed, since a specific SpatialLocalizer won't know about other localizers.